File: 009 Panlidrome Number/PalindromeNumber.java
class PalindromeNumber {
    public boolean isPalindrome(int x) {
        // special case 
        if (x == 0) {
            return true;
        }
        // other base cases
        else if (x < 0 || x % 10 == 0) {
            return false;
        }
        
        // reverse the integer
        int reverse = 0;
        int curr;
        int num = x;
        while (num != 0 ) {
            curr = num % 10;
            num = num / 10;
            reverse = reverse * 10 + curr;
        }
        
        return reverse == x;
    }
}

File: 9 Panlidrome Number/palindrome_number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x == 0:
            return True
        elif x < 0 or x % 10 == 0:
            return False
        
        num = x
        reverse = 0
        while (num != 0):
            curr = num % 10
            num = num // 10
            reverse = reverse * 10 + curr
        return reverse == x